provider/datagen/src/transform/cldr/displaynames/mod.rs

```diff
@@ -0,0 +1,77 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#![cfg(feature = "experimental")]
+
+use crate::transform::cldr::cldr_serde;
+use icu_displaynames::provider::*;
+use icu_provider::datagen::IterableDataProvider;
+use icu_provider::prelude::*;
+use zerovec::ZeroMap;
+
+impl DataProvider<DisplayNamesV1Marker> for crate::DatagenProvider {
+    fn load(&self, req: DataRequest) -> Result<DataResponse<DisplayNamesV1Marker>, DataError> {
+        let langid = req.locale.get_langid();
+
+        let data: &cldr_serde::displaynames::Resource = self
+            .source
+            .cldr()?
+            .displaynames()
+            .read_and_parse(&langid, "territories.json")?;
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(DisplayNamesV1::from(data))),
+        })
+    }
+}
+
+impl IterableDataProvider<DisplayNamesV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(vec![Default::default()])
```

Comment:
This should return all of the locales that are supported in the CLDR input. See the decimal, time zone, etc., transformers for examples of how to impl this function.

Reply:
Done.